File: app/src/main/java/com/silentgames/silent_planet/model/SpaceCell.java
package com.silentgames.silent_planet.model;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;

import com.silentgames.silent_planet.R;
import com.silentgames.silent_planet.utils.BitmapEditor;
import com.silentgames.silent_planet.utils.Converter;

/**
 * Created by gidroshvandel on 07.07.16.
 */
public class SpaceCell {

    private static Bitmap bitmap ;

    private static String properties;

    public SpaceCell() {

    }

    public static Bitmap getBitmap(Resources res, Context context) {
        bitmap = BitmapFactory.decodeResource(res, R.drawable.space_texture);
        bitmap = BitmapEditor.resize(bitmap, Converter.convertDpToPixel(34,context),Converter.convertDpToPixel(34,context));
        return bitmap;
    }

    public static String getProperties() {
        return properties;
    }

    public static void setProperties(String properties) {
        SpaceCell.properties = properties;
    }
}

File: app/src/main/java/com/silentgames/silent_planet/model/cells/SpaceCell.kt
package com.silentgames.silent_planet.model.cells

import android.content.Context
import android.graphics.Bitmap
import com.silentgames.silent_planet.R
import com.silentgames.silent_planet.model.GameMatrixHelper
import com.silentgames.silent_planet.utils.BitmapEditor

/**
 * Created by gidroshvandel on 07.07.16.
 */
class SpaceCell(
        context: Context,
        override var bitmap: Bitmap = BitmapEditor.getCellBitmap(context, R.drawable.space_texture),
        override var isCanMove: Boolean = false
) : CellType(
        context,
        isCanFly = true,
        closeBitmap = BitmapEditor.getCellBitmap(context, R.drawable.space_texture),
        name = context.getString(R.string.space_cell_name),
        description = context.getString(R.string.space_cell_description)
) {

    override fun doEvent(gameMatrixHelper: GameMatrixHelper): GameMatrixHelper {
        return gameMatrixHelper
    }
}
